k8s/eks/bash/functions.sh

```diff
@@ -0,0 +1,312 @@
+    kubectl apply -f ../kops/efs/rbac.yaml -f $EFS_MANIFEST_SPEC
+}
+
+aws_create_ebs(){
+  create_ebs_volume=$(aws ec2 create-volume --size $EBS_SIZE  --availability-zone $AVAILABILITY_ZONE)
+  echo "$create_ebs_volume"
+  ebs_volume=$(echo $create_ebs_volume | jq -r '.VolumeId' )
+  echo "ebs=$ebs_volume" >> ./.env
+}
+
+make_persistent_volume(){  
+export TG_EBS_DATADIR_VOLUME_ID=$ebs_volume
+EBS_PV=$(mktemp)
+envsubst <../kops/ebs/pv.yml.spec >$EBS_PV
+kubectl apply -f ../kops/ebs/storageclass.yml -f $EBS_PV -f ../kops/ebs/pvc.yml
+}
+
+helm_redis_add_repo(){
+  helm repo add bitnami https://charts.bitnami.com/bitnami
+} 
+
+helm_infra_install_redis(){
+  helm install testground-infra-redis --set auth.enabled=false --set master.nodeSelector='testground.node.role.infra: "true"' bitnami/redis
+} 
+
+# Until we find a better way for region var
+# tg_daemon_config_map(){
+#   kubectl apply -f ../kops/testground-daemon/config-map-env-toml.yml
+# } | tee -a ./log/$start-log/tg_daemon_config_map.log
+
+tg_daemon_config_map(){
+  kubectl create -f - <<EOF
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: env-toml-cfg
+  namespace: default
+data:
+  .env.toml: |
+    ["aws"]
+    region = "$REGION"   
+
+    [runners."cluster:k8s"]
+    run_timeout_min             = 15
+    testplan_pod_cpu            = "100m"
+    testplan_pod_memory         = "100Mi"
+    collect_outputs_pod_cpu     = "1000m"
+    collect_outputs_pod_memory  = "1000Mi"
+    provider                    = "aws"
+    sysctls = [
+      "net.core.somaxconn=10000",
+    ]
+
+    [daemon]
+    listen = "0.0.0.0:8042"
+    slack_webhook_url = ""
+    github_repo_status_token = ""
+
+    [daemon.scheduler]
+    workers = 2
+    task_timeout_min  = 20
+    task_repo_type    = "disk"
+
+    [client]
+    endpoint = "localhost:8080"
+
+EOF
+}
+
+tg_daemon_service_account(){
+  kubectl apply -f ./yaml/tg-daemon-service-account.yml
+}
+
+tg_daemon_role_binding(){
+  kubectl apply -f ./yaml/tg-daemon-cluster-role.yml
+}
+
+tg_daemon_services(){
+  kubectl apply -f ./yaml/tg-testground-daemon-service.yml
+}
+
+tg_daemon_svc_sync_service(){
+  kubectl apply -f ./yaml/tg-sync-service.yml
+}
+
+tg_daemon_sync_service(){
+  kubectl apply -f ./yaml/tg-sync-service-deployment.yml
+}
+
+tg_daemon_sidecar(){
+  kubectl apply -f ./yaml/tg-ds-sidecar.yml
+}
+
+tg_daemon_deployment(){
+  kubectl apply -f ./yaml/tg-daemon-deployment.yml
+}
+
+log(){
+  tar czf ./log/$start-$CLUSTER_NAME-$CNI_COMBINATION.tar.gz /log/$start-log/
+  echo "##########################################"
+  echo "Log file generated with name $start-$CLUSTER_NAME-$CNI_COMBINATION.tar.gz"
+  echo "##########################################"
+  rm -rf ./log/$start-log/ 
+}
+
+# cluster_creation_manifest(){
+#   mkdir -p .cluster_manifest
+#   echo "$CLUSTER_NAME" >> clusters
+# }
+
+cleanup(){
+  mnt_target_id=$(aws efs describe-mount-targets --file-system-id $efs | jq -r ".MountTargets[] | .MountTargetId")
```

Comment:
I believe `$efs` value was never set

Reply:
Since the script did not run from the `eks` folder, it couldn't see the necessary files, so efs probably couldn't be created. 
I believe paths need to be fixed, so it doesn't matter where you run the script from.

Reply:
That is not correct, the script fails immediately (line 4) when it's run in a different location (I assumed you tried it),

This is a different issue.

Reply:
Exactly, the script fails at the beginning and it doesn't even execute the part where an EFS is created.
If an EFS was created, it would store the `$efs` value and would know to refer to it while performing the cleanup.

But since it does not have an EFS created, it is also missing the `$efs` value you noted here.